crates/sui-config/src/genesis.rs

```diff
@@ -544,19 +541,11 @@ impl Builder {
             .map(|genesis_info| &genesis_info.info)
             .zip(system_object.validators.active_validators.iter())
         {
-            assert_eq!(
-                validator.sui_address().to_vec(),
-                onchain_validator.metadata.sui_address.to_vec(),
-            );
-            assert_eq!(
-                validator.protocol_key().as_ref().to_vec(),
-                onchain_validator.metadata.protocol_pubkey_bytes,
-            );
-            assert_eq!(validator.name(), onchain_validator.metadata.name);
-            assert_eq!(
-                validator.network_address().to_vec(),
-                onchain_validator.metadata.net_address
-            );
+            let metadata = onchain_validator.verified_metadata();
+            assert_eq!(validator.sui_address(), metadata.sui_address);
+            assert_eq!(validator.protocol_key(), metadata.sui_pubkey_bytes);
```

Comment:
A bit confusing why we are checking protocol_key with sui_pubkey_bytes, maybe rename one of them?